File: modules/coremodules/auth/src/main/java/com/example/auth/entities/Privilege.java
package com.example.auth.entities;

import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "privileges", schema = "auth")
public class Privilege extends BaseEntity {

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false, unique = true)
    private String label;

    private String description;

    @ElementCollection
    @LazyCollection(LazyCollectionOption.FALSE)
    @CollectionTable(name = "privileges_access_urls", schema = "auth")
    private List<String> accessUrls;


    public Privilege() {
    }

    public Privilege(String name, String label) {
        this.name = name;
        this.label = label;
    }

    public String[] accessesArr() {
        String[] itemsArray = new String[accessUrls.size()];
        return accessUrls.toArray(itemsArray);
    }

    public String accessesStr() {
        return String.join(",", accessUrls);
    }

    public String getName() {
        return name;
    }

    public String getLabel() {
        return label;
    }

    public String getDescription() {
        return description;
    }

    public List<String> getAccessUrls() {
        return accessUrls;
    }

    public void setAccessUrls(List<String> accessUrls) {
        this.accessUrls = accessUrls;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public void setDescription(String description) {
        this.description = description;
    }


}

File: src/main/kotlin/com/example/webservice/domains/users/models/entities/Privilege.kt
package com.example.webservice.domains.users.models.entities

import com.example.webservice.domains.common.models.entities.base.BaseEntity
import org.hibernate.annotations.LazyCollection
import org.hibernate.annotations.LazyCollectionOption
import java.util.*
import javax.persistence.*

import kotlin.collections.ArrayList

@Entity
@Table(name = "privileges")
class Privilege() : BaseEntity() {
    @Column(nullable = false, unique = true)
    lateinit var name: String

    @Column(nullable = false, unique = true)
    lateinit var label: String

    @ElementCollection
    @LazyCollection(LazyCollectionOption.FALSE)
    @CollectionTable(name = "privileges_access_urls")
    lateinit var accessUrls: List<String>

    fun accessesArr(): Array<String> {
        return accessUrls.toTypedArray()
    }

    constructor(name: String, label: String) : this() {
        this.name = name
        this.label = label
    }

    enum class Privileges(val label: String) {
        ADMINISTRATION("Administration"), ACCESS_USER_RESOURCES("Access User Resources")
    }

    public fun accessesStr(): String {
        return this.accessUrls.joinToString()
    }
}
